### ISSUE TITLE
WinMerge declared "out of date"

### ISSUE THREAD

**Issue by Taylorpsepa** [WinMerge/winmerge#2412]
I installed the latest version of WinMerge, 2.16.42.1, on my computer. I have two software packages installed, Vipre AntiVirus, an CCleaner. Both of those packages monitor my installed software, looking for out-of-date software. They both say that the current version of WinMerge is 2.16.42.1, but that my version is out-of-date - yet, if I open WinMerge, go to Help|About, I am told that I have 2.16.42.1 installed. Both Vipre and CCleaner say that my installed version is 2.16.40.0. Why is that happening?

**Comment by Taylorpsepa** [comment]
I forgot: I have Windows 11 x64 installed on my computer.

**Comment by sdottaka** [comment]
You may have an older 32-bit version of WinMerge installed (C:/Program Files(x86)/WinMerge) or the WinMerge executable file may be stored in another folder.

**Comment by Taylorpsepa** [comment]
Hello,

 

Here is what happened. I found the older version, and uninstalled it. Of course, the uninstall of the older version wiped out part of the installation of the newer version; thus, I had to reinstall the newer version. When I did that, I lost the names and locations of the files that I last had compared, but, more importantly, I lost the plugin information, so I had to restore that (and it’s complicated when the plugin isn’t “mainstream”)..

 

I think that this happened because your installation software for the 64-bit new version did not uninstall the older version. Either your instructions should state that the older version must be manually uninstalled first, or your installation software should uninstall the older version.

 

Paul Taylor

**Comment by sdottaka** [comment]
Was version 2.16.40 of WinMerge the 32-bit version (C:\Program Files(x86)\WinMerge)?

Normally, if you have an older 64-bit version of WinMerge installed and you install a newer 64-bit version of WinMerge, it will be overwritten.

**Comment by Taylorpsepa** [comment]
Yes, I believe that the older version was the 32-bit version. I still think that the 64-bit installer should search for any 32-bit version and uninstall it.

**Comment by FaceCrap** [comment]
> Yes, I believe that the older version was the 32-bit version. I still think that the 64-bit installer should search for any 32-bit version and uninstall it.

Having run ino an almost similar situation with another program, I think this is unadvisable. 
For one thing, non-default plugins for the 32bit version won't automatically get upgraded to a 64bit version. 
Which would have landed you in the same predicament as when you uninstalled it manually.
